rlclientlib/vw_model/vw_model.cc

```diff
@@ -41,27 +40,9 @@ namespace reinforcement_learning { namespace model_management {
       // Get a ranked list of action_ids and corresponding pdf
       vw->rank(features, action_ids, pdf);
 
-      // Pick a slot using the pdf. NOTE: sample_after_normalizing() can change the pdf
-      uint32_t chosen_index;
-      auto scode = e::sample_after_normalizing(rnd_seed, std::begin(pdf), std::end(pdf), chosen_index);
-
-      if ( S_EXPLORATION_OK != scode ) {
-        RETURN_ERROR_LS(_trace_logger, status, exploration_error) << scode;
-      }
-
-      // Setup response with pdf from prediction and action indexes
-      for ( size_t idx = 0; idx < pdf.size(); ++idx ) {
-        response.push_back(action_ids[idx], pdf[idx]);
-      }
-
-      // Swap values in first position with values in chosen index
-      scode = e::swap_chosen(std::begin(response), std::end(response), chosen_index);
-
-      if ( S_EXPLORATION_OK != scode ) {
-        RETURN_ERROR_LS(_trace_logger, status, exploration_error) << "Exploration error code: " << scode;
-      }
-
-      response.set_chosen_action_id(action_ids[chosen_index]);
+      // TODO: Should there be an i_pdf_model? It seems like the ability to integrate at the PDF level easily would be nice.
```

Comment:
Agree.  

I think current i_model does not work well.  The following change makes sense to me.

1) i_model that only returns collection of (actions, probabilities)
2) not sure if an i_sampler is needed simply the sampling functionality moved out of i_model.

I meant to go back and fix this debt.

I would be all for changing i_model now with this PR since no one else is using it and we are promoting a bad pattern that others are likely to mimic.